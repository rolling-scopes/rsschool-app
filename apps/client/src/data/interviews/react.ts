import { InterviewTemplate } from './types';

export const reactTemplate: InterviewTemplate = {
  name: 'React interview ',
  examplesUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/react/questions.md',
  categories: [
    {
      id: 1000,
      name: 'JSX',
      questions: [
        { id: 1001, name: 'What is JSX?' },
        { id: 1002, name: 'Is it possible to use React without JSX?' },
      ],
    },
    {
      id: 1010,
      name: 'React DOM',
      questions: [
        { id: 1011, name: 'What is the virtual DOM? How does react use the virtual DOM to render the UI?' },
        { id: 1012, name: 'Is the virtual DOM the same as the shadow DOM?' },
        { id: 1013, name: 'What is the difference between the virtual DOM and the real DOM?' },
      ],
    },
    {
      id: 1020,
      name: 'Render',
      questions: [
        { id: 1021, name: 'When is a component rendered?' },
        { id: 1022, name: 'How not to render on props change?' },
        { id: 1023, name: 'Is it OK to use arrow functions in render methods?' },
      ],
    },
    {
      id: 1030,
      name: 'Interaction between components',
      questions: [
        { id: 1031, name: 'How do you pass a value from parent to child?' },
        { id: 1032, name: 'How do you pass a value from child to parent?' },
        { id: 1033, name: 'What is prop drilling?' },
        { id: 1034, name: 'Can a child component modify its own props?' },
        { id: 1035, name: 'How do you pass a value from sibling to sibling?' },
      ],
    },
    {
      id: 1040,
      name: 'Lifecycle and State',
      questions: [
        { id: 1041, name: 'What is the difference between props and state?' },
        { id: 1042, name: 'How does state in a class component differ from state in a functional component?' },
        { id: 1043, name: 'What is the component lifecycle?' },
        { id: 1044, name: 'How do you update lifecycle in function components?' },
      ],
    },
    {
      id: 1050,
      name: 'Ref',
      questions: [
        { id: 1051, name: 'What is the difference between refs and state variables?' },
        { id: 1052, name: 'When is the best time to use refs?' },
        { id: 1053, name: 'What is the proper way to update a ref in a function component?' },
      ],
    },
    {
      id: 1060,
      name: 'Context',
      questions: [
        { id: 1061, name: 'What is the difference between the context API and prop drilling?' },
        { id: 1062, name: "When shouldn't you use the context API?" },
      ],
    },
    {
      id: 1070,
      name: 'Redux',
      questions: [
        { id: 1071, name: 'Enumerate base principles' },
        { id: 1072, name: 'What is the typical flow of data in a React + Redux app?' },
        { id: 1073, name: 'Benefits of Redux?' },
      ],
    },
    {
      id: 1080,
      name: 'Other',
      questions: [
        { id: 1081, name: 'Is it a good idea to use Math.random for keys?' },
        { id: 1082, name: 'What are the limitations of React?' },
        { id: 1083, name: 'What is a higher order component?' },
        { id: 1084, name: 'What are uncontrolled and controlled components?' },
        { id: 1085, name: 'React optimizations' },
      ],
    },
    {
      id: 1100,
      name: 'Coding task',
      questions: [{ id: 1101, name: 'Small react app: form, button, results list' }],
    },
  ],
};
