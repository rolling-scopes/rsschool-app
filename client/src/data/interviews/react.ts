import { InterviewTemplate } from './types';

export const reactTemplate: InterviewTemplate = {
  name: 'React interview',
  examplesUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/react/questions.md',
  categories: [
    {
      id: 1000,
      name: 'JSX',
      questions: [
        { id: 1001, name: 'What is JSX?' },
        { id: 1002, name: 'Is it possible to use React without JSX?' },
      ],
    },
    {
      id: 1010,
      name: 'Styled-components',
      questions: [{ id: 1011, name: 'Using styled-components' }],
    },
    {
      id: 1020,
      name: 'Components',
      questions: [
        { id: 1021, name: 'Component Definition (Class-based, Functional)' },
        { id: 1022, name: 'Components Composition' },
      ],
    },
    {
      id: 1030,
      name: 'React DOM',
      questions: [
        { id: 1031, name: 'What is the virtual DOM? How does react use the virtual DOM to render the UI?' },
        { id: 1032, name: 'Is the virtual DOM the same as the shadow DOM?' },
        { id: 1033, name: 'What is the difference between the virtual DOM and the real DOM?' },
      ],
    },
    {
      id: 1040,
      name: 'Render',
      questions: [
        { id: 1041, name: 'When is a component rendered?' },
        { id: 1042, name: 'How not to render on props change?' },
        { id: 1043, name: 'Is it OK to use arrow functions in render methods?' },
        { id: 1044, name: 'Components rendering system' },
      ],
    },
    {
      id: 1050,
      name: 'Interaction between components',
      questions: [
        { id: 1051, name: 'How do you pass a value from parent to child?' },
        { id: 1052, name: 'How do you pass a value from child to parent?' },
        { id: 1053, name: 'What is prop drilling?' },
        { id: 1054, name: 'Can a child component modify its own props?' },
        { id: 1055, name: 'How do you pass a value from sibling to sibling?' },
      ],
    },
    {
      id: 1060,
      name: 'Lifecycle and State',
      questions: [
        { id: 1061, name: 'What is the difference between props and state?' },
        { id: 1062, name: 'How does state in a class component differ from state in a functional component?' },
        { id: 1063, name: 'What is the component lifecycle?' },
        { id: 1064, name: 'How do you update lifecycle in function components?' },
        { id: 1065, name: 'Controlled/uncontrolled components' },
        { id: 1066, name: 'Stateful vs stateless components' },
      ],
    },
    {
      id: 1070,
      name: 'Ref',
      questions: [
        { id: 1071, name: 'What is the difference between refs and state variables?' },
        { id: 1072, name: 'When is the best time to use refs?' },
        { id: 1073, name: 'What is the proper way to update a ref in a function component?' },
      ],
    },
    {
      id: 1080,
      name: 'Context',
      questions: [
        { id: 1081, name: 'What is the difference between the context API and prop drilling?' },
        { id: 1082, name: "When shouldn't you use the context API?" },
      ],
    },
    {
      id: 1090,
      name: 'Redux',
      questions: [
        { id: 1091, name: 'Enumerate base principles' },
        { id: 1092, name: 'What is the typical flow of data in a React + Redux app?' },
        { id: 1093, name: 'Benefits of Redux?' },
        { id: 1094, name: 'Async Redux flow' },
      ],
    },
    {
      id: 1100,
      name: 'State Management by managers',
      questions: [
        { id: 1101, name: 'Mobx' },
        { id: 1102, name: 'Reflux' },
      ],
    },
    {
      id: 1110,
      name: 'Routing',
      questions: [
        { id: 1111, name: 'React Router' },
        { id: 1112, name: 'History' },
      ],
    },
    {
      id: 1120,
      name: 'Other',
      questions: [
        { id: 1121, name: 'Is it a good idea to use Math.random for keys?' },
        { id: 1122, name: 'What are the limitations of React?' },
        { id: 1123, name: 'What is a higher order component?' },
        { id: 1124, name: 'What are uncontrolled and controlled components?' },
        { id: 1125, name: 'React optimizations' },
      ],
    },
    {
      id: 1130,
      name: 'Coding task',
      questions: [{ id: 1131, name: 'Small react app: form, button, results list' }],
    },
    {
      id: 1140,
      name: 'Next.js',
      questions: [
        { id: 1141, name: 'What is Next.js, and how does it differ from traditional React applications?' },
        { id: 1142, name: 'Explain the concept of server-side rendering (SSR) in Next.js' },
        { id: 1143, name: 'How does Next.js handle client-side routing?' },
        { id: 1144, name: 'What are the benefits of using Next.js for building React applications?' },
        {
          id: 1145,
          name: 'What is the purpose of the getInitialProps function in Next.js? How is it different from using getStaticProps or getServerSideProps?',
        },
        { id: 1146, name: 'How does Next.js handle automatic code splitting, and why is it important?' },
        {
          id: 1147,
          name: 'What is the purpose of the Link component in Next.js, and how does it differ from traditional anchor (<a>) tags?',
        },
        { id: 1148, name: 'How can you configure routing in a Next.js application?' },
      ],
    },
    {
      id: 1150,
      name: 'Data Fetching',
      questions: [
        {
          id: 1151,
          name: 'Compare and contrast getStaticProps and getServerSideProps in terms of use cases and performance considerations',
        },
        {
          id: 1152,
          name: 'When would you choose to use getStaticPaths in Next.js, and how does it relate to dynamic routes?',
        },
      ],
    },
    {
      id: 1160,
      name: 'Advanced Topics',
      questions: [
        { id: 1161, name: 'Describe the purpose and use cases of API routes in Next.js' },
        { id: 1162, name: 'Explain the concept of middleware in Next.js and its role in the application lifecycle' },
        { id: 1163, name: 'How does Next.js handle authentication in applications?' },
      ],
    },
  ],
};
