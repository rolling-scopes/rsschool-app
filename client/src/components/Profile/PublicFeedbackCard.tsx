import * as React from 'react';
import isEqual from 'lodash/isEqual';
import moment from 'moment';
import { Typography, Comment, Tooltip, Avatar, Badge } from 'antd';
import CommonCard from './CommonCard';
import PublicFeedbackModal from './PublicFeedbackModal';
import heroesBadges from '../../configs/heroes-badges';
import { PublicFeedback } from '../../../../common/models/profile';
import { ConfigurableProfilePermissions } from '../../../../common/models/profile';
import { ChangedPermissionsSettings } from 'pages/profile';
import { CheckboxChangeEvent } from 'antd/lib/checkbox';

const { Text, Paragraph } = Typography;

import { MessageOutlined, FullscreenOutlined } from '@ant-design/icons';

type Props = {
  data: PublicFeedback[];
  isEditingModeEnabled: boolean;
  permissionsSettings?: ConfigurableProfilePermissions;
  onPermissionsSettingsChange: (event: CheckboxChangeEvent, settings: ChangedPermissionsSettings) => void;
};

interface State {
  badgesCount: {
    [key: string]: number;
  };
  isPublicFeedbackModalVisible: boolean;
}

class PublicFeedbackCard extends React.Component<Props, State> {
  state: State = {
    badgesCount: {},
    isPublicFeedbackModalVisible: false,
  };

  private filterPermissions = ({ isPublicFeedbackVisible }: Partial<ConfigurableProfilePermissions>) => ({
    isPublicFeedbackVisible,
  });

  private showPublicFeedbackModal = () => {
    this.setState({ isPublicFeedbackModalVisible: true });
  };

  private hidePublicFeedbackModal = () => {
    this.setState({ isPublicFeedbackModalVisible: false });
  };

  private countBadges = () => {
    const receivedBadges = this.props.data;
    const badgesCount: any = {};

    receivedBadges.forEach(({ badgeId }) => {
      if (badgeId) {
        badgesCount[badgeId] ? (badgesCount[badgeId] += 1) : (badgesCount[badgeId] = 1);
      }
    });

    return badgesCount;
  };

  shouldComponentUpdate = (nextProps: Props, nextState: State) =>
    !isEqual(
      nextProps.permissionsSettings?.isPublicFeedbackVisible,
      this.props.permissionsSettings?.isPublicFeedbackVisible,
    ) ||
    !isEqual(nextProps.isEditingModeEnabled, this.props.isEditingModeEnabled) ||
    !(nextState.isPublicFeedbackModalVisible === this.state.isPublicFeedbackModalVisible) ||
    !isEqual(nextState.badgesCount, this.state.badgesCount);

  componentDidMount() {
    const badgesCount = this.countBadges();
    this.setState({ badgesCount });
  }

  render() {
    const { isEditingModeEnabled, permissionsSettings, onPermissionsSettingsChange } = this.props;
    const badges = this.props.data;
    const { badgesCount } = this.state;

    return (
      <>
        <PublicFeedbackModal
          data={badges}
          isVisible={this.state.isPublicFeedbackModalVisible}
          onHide={this.hidePublicFeedbackModal}
        />
        <CommonCard
          title="Public Feedback"
          icon={<MessageOutlined />}
          actions={[
            <FullscreenOutlined key="card-public-feedback-button-more" onClick={this.showPublicFeedbackModal} />,
          ]}
          content={
            <>
              <div style={{ marginBottom: 20 }}>
                <Text strong>Total badges:</Text> {badges.length}
              </div>
              <div style={{ marginBottom: 30 }}>
                {Object.keys(badgesCount).map(badgeId => (
                  <div style={{ margin: 5, display: 'inline-block' }} key={`badge-${badgeId}`}>
                    <Badge count={badgesCount[badgeId]}>
                      <Tooltip title={(heroesBadges as any)[badgeId].name}>
                        <Avatar
                          src={`/static/svg/badges/${(heroesBadges as any)[badgeId].url}`}
                          alt={`${badgeId} badge`}
                          size={48}
                        />
                      </Tooltip>
                    </Badge>
                  </div>
                ))}
              </div>
              <div style={{ marginBottom: 0 }}>
                <Text strong>Last feedback:</Text>
              </div>
              {badges.slice(0, 1).map(({ fromUser, comment, feedbackDate, badgeId }, idx) => (
                <Comment
                  key={`comment-${idx}`}
                  author={<a href={`/profile?githubId=${fromUser.githubId}`}>{fromUser.name}</a>}
                  avatar={
                    <Avatar
                      src={`https://github.com/${fromUser.githubId}.png?size=${48}`}
                      alt={`${fromUser.githubId} avatar`}
                    />
                  }
                  content={
                    <>
                      {badgeId ? (
                        <Text strong style={{ fontSize: 12 }}>
                          {(heroesBadges as any)[badgeId].name}
                        </Text>
                      ) : (
                        ''
                      )}
                      <Paragraph ellipsis={{ rows: 3, expandable: true }}>{comment}</Paragraph>
                    </>
                  }
                  datetime={
                    <Tooltip title={moment(feedbackDate).format('YYYY-MM-DD HH:mm:ss')}>
                      <span>{moment(feedbackDate).fromNow()}</span>
                    </Tooltip>
                  }
                />
              ))}
            </>
          }
          permissionsSettings={permissionsSettings ? this.filterPermissions(permissionsSettings) : undefined}
          isEditingModeEnabled={isEditingModeEnabled}
          onPermissionsSettingsChange={onPermissionsSettingsChange}
        />
      </>
    );
  }
}

export default PublicFeedbackCard;
