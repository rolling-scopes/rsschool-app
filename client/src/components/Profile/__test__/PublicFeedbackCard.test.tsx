import React from 'react';
import { render } from '@testing-library/react';
import PublicFeedbackCard from '../PublicFeedbackCard';

jest.mock('moment', () => (value: any) => ({
  format() {
    return value;
  },
  fromNow() {
    return 'fromNow';
  },
}));

describe('PublicFeedbackCard', () => {
  const data = [
    {
      feedbackDate: '2018-12-01T12:12:01.000Z',
      badgeId: 'Congratulations',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Anton Petrov',
        githubId: 'apetr',
      },
    },
    {
      feedbackDate: '2018-11-01T11:12:01.000Z',
      badgeId: 'Great_speaker',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Artem Petrov',
        githubId: 'temap',
      },
    },
    {
      feedbackDate: '2018-09-01T11:12:01.000Z',
      badgeId: 'Great_speaker',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Artem Petrov',
        githubId: 'temap',
      },
    },
    {
      feedbackDate: '2018-10-01T11:12:01.000Z',
      badgeId: 'Great_speaker',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Artem Petrov',
        githubId: 'temap',
      },
    },
    {
      feedbackDate: '2018-11-01T12:12:01.000Z',
      badgeId: 'Thank_you',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Anton Vasilyev',
        githubId: 'vasssa',
      },
    },
    {
      feedbackDate: '2019-12-01T12:12:01.000Z',
      badgeId: 'Thank_you',
      comment: 'Test',
      heroesUri: 'https://heroes.by/',
      fromUser: {
        name: 'Dima Alexandrov',
        githubId: 'demaa',
      },
    },
  ];

  it('should render correctly', () => {
    const output = render(<PublicFeedbackCard data={data} />);
    expect(output.container).toMatchSnapshot();
  });
});
