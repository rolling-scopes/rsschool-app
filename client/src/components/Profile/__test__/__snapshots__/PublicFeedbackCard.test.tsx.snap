// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicFeedbackCard Should render correctly if is editing mode disabled 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <h2
            class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line"
            style="font-size: 16px; margin-bottom: 0px;"
          >
            <span
              aria-label="message"
              class="anticon anticon-message"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="message"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                />
              </svg>
            </span>
             
            Public Feedback
          </h2>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        style="margin-bottom: 20px;"
      >
        <span
          class="ant-typography"
        >
          <strong>
            Total badges:
          </strong>
        </span>
         
        6
      </div>
      <div
        style="margin-bottom: 30px;"
      >
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Congratulations badge"
                src="/static/svg/badges/Congratulations.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="1"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  1
                </span>
              </span>
            </sup>
          </span>
        </div>
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Great_speaker badge"
                src="/static/svg/badges/GreatSpeaker.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="3"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  3
                </span>
              </span>
            </sup>
          </span>
        </div>
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Thank_you badge"
                src="/static/svg/badges/ThankYou.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="2"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  2
                </span>
              </span>
            </sup>
          </span>
        </div>
      </div>
      <div
        style="margin-bottom: 0px;"
      >
        <span
          class="ant-typography"
        >
          <strong>
            Last feedback:
          </strong>
        </span>
      </div>
      <div
        class="ant-comment"
      >
        <div
          class="ant-comment-inner"
        >
          <div
            class="ant-comment-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                src="https://cdn.rs.school/apetr.png?size=96"
              />
            </span>
          </div>
          <div
            class="ant-comment-content"
          >
            <div
              class="ant-comment-content-author"
            >
              <span
                class="ant-comment-content-author-name"
              >
                <a
                  href="/profile?githubId=apetr"
                >
                  Anton Petrov
                </a>
              </span>
              <span
                class="ant-comment-content-author-time"
              >
                <span>
                  fromNow
                </span>
              </span>
            </div>
            <div
              class="ant-comment-content-detail"
            >
              <span
                class="ant-typography"
                style="font-size: 12px;"
              >
                <strong>
                  Congratulations
                </strong>
              </span>
              <div
                aria-label="Test"
                class="ant-typography ant-typography-ellipsis"
              >
                Test
                <span
                  aria-hidden="true"
                  style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;"
                >
                  lg
                </span>
                <span
                  aria-hidden="true"
                  style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
                >
                  <span
                    aria-hidden="true"
                  >
                    ...
                  </span>
                  <a
                    aria-label="Expand"
                    class="ant-typography-expand"
                  >
                    Expand
                  </a>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="ant-card-actions"
    >
      <li
        style="width: 100%;"
      >
        <span>
          <span
            aria-label="fullscreen"
            class="anticon anticon-fullscreen"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="fullscreen"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M290 236.4l43.9-43.9a8.01 8.01 0 00-4.7-13.6L169 160c-5.1-.6-9.5 3.7-8.9 8.9L179 329.1c.8 6.6 8.9 9.4 13.6 4.7l43.7-43.7L370 423.7c3.1 3.1 8.2 3.1 11.3 0l42.4-42.3c3.1-3.1 3.1-8.2 0-11.3L290 236.4zm352.7 187.3c3.1 3.1 8.2 3.1 11.3 0l133.7-133.6 43.7 43.7a8.01 8.01 0 0013.6-4.7L863.9 169c.6-5.1-3.7-9.5-8.9-8.9L694.8 179c-6.6.8-9.4 8.9-4.7 13.6l43.9 43.9L600.3 370a8.03 8.03 0 000 11.3l42.4 42.4zM845 694.9c-.8-6.6-8.9-9.4-13.6-4.7l-43.7 43.7L654 600.3a8.03 8.03 0 00-11.3 0l-42.4 42.3a8.03 8.03 0 000 11.3L734 787.6l-43.9 43.9a8.01 8.01 0 004.7 13.6L855 864c5.1.6 9.5-3.7 8.9-8.9L845 694.9zm-463.7-94.6a8.03 8.03 0 00-11.3 0L236.3 733.9l-43.7-43.7a8.01 8.01 0 00-13.6 4.7L160.1 855c-.6 5.1 3.7 9.5 8.9 8.9L329.2 845c6.6-.8 9.4-8.9 4.7-13.6L290 787.6 423.7 654c3.1-3.1 3.1-8.2 0-11.3l-42.4-42.4z"
              />
            </svg>
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PublicFeedbackCard Should render correctly if is editing mode enabled 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <h2
            class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line"
            style="font-size: 16px; margin-bottom: 0px;"
          >
            <span
              aria-label="message"
              class="anticon anticon-message"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="message"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M464 512a48 48 0 1096 0 48 48 0 10-96 0zm200 0a48 48 0 1096 0 48 48 0 10-96 0zm-400 0a48 48 0 1096 0 48 48 0 10-96 0zm661.2-173.6c-22.6-53.7-55-101.9-96.3-143.3a444.35 444.35 0 00-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6.3-119.3 12.3-174.5 35.9a445.35 445.35 0 00-142 96.5c-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9A449.4 449.4 0 00112 714v152a46 46 0 0046 46h152.1A449.4 449.4 0 00510 960h2.1c59.9 0 118-11.6 172.7-34.3a444.48 444.48 0 00142.8-95.2c41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5.3-60.9-11.5-120-34.8-175.6zm-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-.6 99.6-39.7 192.9-110.1 262.7z"
                />
              </svg>
            </span>
             
            Public Feedback
          </h2>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        style="margin-bottom: 20px;"
      >
        <span
          class="ant-typography"
        >
          <strong>
            Total badges:
          </strong>
        </span>
         
        6
      </div>
      <div
        style="margin-bottom: 30px;"
      >
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Congratulations badge"
                src="/static/svg/badges/Congratulations.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="1"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  1
                </span>
              </span>
            </sup>
          </span>
        </div>
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Great_speaker badge"
                src="/static/svg/badges/GreatSpeaker.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="3"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  3
                </span>
              </span>
            </sup>
          </span>
        </div>
        <div
          style="margin: 5px; display: inline-block;"
        >
          <span
            class="ant-badge"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                alt="Thank_you badge"
                src="/static/svg/badges/ThankYou.svg"
              />
            </span>
            <sup
              class="ant-scroll-number ant-badge-count"
              data-show="true"
              title="2"
            >
              <span
                class="ant-scroll-number-only"
                style="transition: none;"
              >
                <span
                  class="ant-scroll-number-only-unit current"
                >
                  2
                </span>
              </span>
            </sup>
          </span>
        </div>
      </div>
      <div
        style="margin-bottom: 0px;"
      >
        <span
          class="ant-typography"
        >
          <strong>
            Last feedback:
          </strong>
        </span>
      </div>
      <div
        class="ant-comment"
      >
        <div
          class="ant-comment-inner"
        >
          <div
            class="ant-comment-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
              style="width: 48px; height: 48px; line-height: 48px; font-size: 18px;"
            >
              <img
                src="https://cdn.rs.school/apetr.png?size=96"
              />
            </span>
          </div>
          <div
            class="ant-comment-content"
          >
            <div
              class="ant-comment-content-author"
            >
              <span
                class="ant-comment-content-author-name"
              >
                <a
                  href="/profile?githubId=apetr"
                >
                  Anton Petrov
                </a>
              </span>
              <span
                class="ant-comment-content-author-time"
              >
                <span>
                  fromNow
                </span>
              </span>
            </div>
            <div
              class="ant-comment-content-detail"
            >
              <span
                class="ant-typography"
                style="font-size: 12px;"
              >
                <strong>
                  Congratulations
                </strong>
              </span>
              <div
                aria-label="Test"
                class="ant-typography ant-typography-ellipsis"
              >
                Test
                <span
                  aria-hidden="true"
                  style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;"
                >
                  lg
                </span>
                <span
                  aria-hidden="true"
                  style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
                >
                  <span
                    aria-hidden="true"
                  >
                    ...
                  </span>
                  <a
                    aria-label="Expand"
                    class="ant-typography-expand"
                  >
                    Expand
                  </a>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-drawer ant-drawer-top ant-drawer-inline"
        style="position: absolute; display: none;"
        tabindex="-1"
      >
        <div
          aria-hidden="true"
          data-sentinel="start"
          style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
          tabindex="0"
        />
        <div
          aria-hidden="true"
          data-sentinel="end"
          style="width: 0px; height: 0px; overflow: hidden; outline: none; position: absolute;"
          tabindex="0"
        />
      </div>
      <style>
        
        :global(.ant-drawer-content-wrapper) {
          height: inherit !important;
        }

        :global(.ant-drawer-wrapper-body) {
          overflow: hidden;
        }

        :global(.ant-drawer-body) {
          height: calc(100% - 55px);
          overflow: auto;
        }
      
      </style>
    </div>
  </div>
</div>
`;
